### PR TITLE
[OpenMP][Flang] Fix dynamic-extent array mapping

### PR DESCRIPTION
**Pull request by TIFitis**
This patch fixes the mapping and lowering of arrays with dynamic extents and adds a new test for the same. The fix discards the incomplete the dynamic extent information and replacing it with just the base type. When lowering to llvm later, the bounds information is used instead.

### CODE REVIEW

**Review by agozillon — APPROVED** [review]
LGTM! Thank you very much for the fix :-)

File: offload/test/offloading/fortran/target-map-dynamic.f90 [hunk 1]
```diff
@@ -0,0 +1,33 @@
+! Offloading test checking lowering of arrays with dynamic extents.
+! REQUIRES: flang, amdgpu
+
+! RUN: %libomptarget-compile-fortran-run-and-check-generic
+
+subroutine test_array_target_enter_data(dims)
+    integer, intent(in) :: dims(2)
+    double precision :: A(2, dims(2))
+    !$omp target enter data map(to: U)
```
Comment (bcornille) [inline offload/test/offloading/fortran/target-map-dynamic.f90:9]: Should this be `map(to: A)` ? I don't see a variable `U`.

Comment (agozillon) [reply offload/test/offloading/fortran/target-map-dynamic.f90:9]: It should be, thank you for the catch! I've done the same mistake a few times and seen it in a few other tests, and now I always use implicit none... :-)

Comment (TIFitis) [reply offload/test/offloading/fortran/target-map-dynamic.f90:9]: Sorry about that, fixed in https://github.com/llvm/llvm-project/commit/a67b6e163528ea547ccbcd4850803f062df66d2e.

**Pull request merged**